lib/stan.js

```diff
@@ -280,17 +290,28 @@ Stan.prototype.createConnection = function() {
       that.nc.publish(reply);
     });
 
+    that.pingInbox = nats.createInbox();
+    that.pingSubscription = that.nc.subscribe(that.pingInbox, function(msg) {
+      processPingResponse(that, msg);
+    });
+
     that.ackSubscription = that.nc.subscribe(that.ackSubject, that.processAck());
 
     var discoverSubject = that.options.discoverPrefix + '.' + that.clusterID;
     //noinspection JSUnresolvedFunction
+    that.connId = nuid.next();
     var req = new proto.ConnectRequest();
     req.setClientId(that.clientID);
     req.setHeartbeatInbox(hbInbox);
+    req.setProtocol(PROTOCOL_ONE);
+    req.setConnId(that.connId);
+    req.setPingInterval(Math.ceil(that.options.pingInterval / 1000));
+    req.setPingMaxout(that.options.pingMaxOut);
 
     // cleanup the connection
     function cleanupConnection(error) {
       that.emit('error', error);
+      cleanupOnClose(that);
```

Comment:
`cleanupOnClose()` should be functionality in Stan.close(), and this should simply call close.

Reply:
Not sure why additional close logic was added here (before your change).